### DISCUSSION TITLE
Unable to use Geocoder instance inside reverse_geocode method

### DISCUSSION THREAD

**Discussion by rsotogar** [geopandas/geopandas#2712] [Q&A]
When trying to parse the geopy.Geocoders.Nominatim instance as the provider argument to the reverse_geocode() method I am getting the following error:

"KeyError: Nomitatim object is not callable"

Alternatively I tried to set up geopy.geocoders.options.set_default_user but it is not being picked up at runtime either. Any idea why this is happening? I am wrapping up this method inside a Pandas UDF like this:

````
@F.pandas_udf(StringType())
def get_country_from_coordinates(latitude_series: pd.Series, longitude_series: pd.Series) -> pd.Series:
    nominatim = Nominatim(timeout = 100, user_agent = "my_geocoder")
    points_array = geopandas.points_from_xy(longitude_series, latitude_series)
    geo_df = geopandas.tools.reverse_geocode(points_array, nominatim)
    address = geo_df.iloc[:,-1]
    country_series = address.str.split(pat = ",", expand = True).iloc[:,-1]
    return country_series

````

**Comment by martinfleis** [comment]
You should pass `Nominatim` class directly and pass any of their keywords alongside. `reverse_geocode` will pass them through.

```py
geo_df = geopandas.tools.reverse_geocode(points_array, Nominatim, timeout = 100, user_agent = "my_geocoder")
```
